MaxText/pyconfig.py

```diff
@@ -149,13 +149,21 @@ def __init__(self, argv: list[str], **kwargs):
     self._validate_env_variables(raw_data_from_yaml)
 
     raw_keys = OrderedDict()
+
     keys_from_env_and_command_line = self._update_from_env_and_command_line(raw_keys, raw_data_from_yaml, argv, **kwargs)
     max_logging.log(
         f"Updating keys from env and command line: {keys_from_env_and_command_line}")
+
     keys_from_model = _HyperParameters.update_model_vars(raw_keys)
     max_logging.log(f"Updating keys from model: {keys_from_model}")
     validate_no_keys_overwritten_twice(keys_from_env_and_command_line, keys_from_model)
 
+    # We initialize the jax distributed system here because it must be done before device backend is initialized.
+    if raw_keys["enable_checkpointing"] and raw_keys["async_checkpointing"] and raw_keys["compile_topology_num_slices"]==-1:
```

Comment:
@gobbleturk I'm really confused that this is spreading but we need to figure out how to make it stop. Should the three of us get together and discuss this?

Reply:
@ZhiyuLi-goog I think this changed a little upstream but I have no objection.